### PR TITLE
IEP-1033 esptool write_flash command in the IDE

### PR DESCRIPTION
**Pull request by sigmaaa**
## Description

added esptool: write_flash command. UI is made based on also esptool command `erase flash`.

Fixes # ([IEP-1033](https://jira.espressif.com:8443/browse/IEP-1033))

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How has this been tested?

Test A:
- create a Non-Volatile Storage (NVS) Read and Write a Value - Example (nvs_rw_value project)
- create a custom partition table
- generate nvs.bin via `NVS Table Editor`
- make small modifications to the nvs_rw_value project to use values specified in nvs.csv
- flash the project
- write bins to flash via a new command
- verify that nvs.bin is flashed via serial monitor

**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:
- ESP-IDF: Write Binary Data to Flash

## Checklist
- [ ] PR Self Reviewed
- [x] Applied Code formatting
- [x] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

## Summary by CodeRabbit

New Features:
- Added the ability to write binary data to the ESP's flash memory using the ESP-IDF Eclipse plugin. This includes a new command "Write Binary Data to Flash" and a user-friendly dialog box for selecting the serial port, binary file path, and offset.

User Interface:
- Enhanced the flash writing functionality with improved error messages, labels, and informational messages.
- Added validation for binary file existence and offset input.

Documentation:
- Updated the README with a new section "Write Binary to Flash" providing detailed instructions on how to use the new feature.

### CODE REVIEW

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/WriteFlashDialog.java [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+			ports = SerialPort.list();
+			espToolCommands = new EspToolCommands();
+		}
+		catch (IOException e)
+		{
+			Logger.log(e);
+		}
+
+		setShellStyle(getShellStyle() | SWT.RESIZE | SWT.MAX);
+	}
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/WriteFlashDialog.java]: fixed

**Review by kolipakakondal — APPROVED** [review]
Thanks for the good work @sigmaaa LGTM

**Comment by kolipakakondal** [comment]
@AndriiFilippov Please verify this

**Comment by kolipakakondal** [comment]
Hi @sigmaaa Can you also documentation for this improvement?

**Comment by sigmaaa** [comment]
> Hi @sigmaaa Can you also documentation for this improvement?

Hi @kolipakakondal, I'm waiting for a new icon for this command and after that will add documentation

**Review by alirana01 — APPROVED** [review]
LGTM!

**Comment by sigmaaa** [comment]
@kolipakakondal, @AndriiFilippov added documentation

**Comment by sigmaaa** [comment]
added new icon

**Comment by AndriiFilippov** [comment]
@sigmaaa hi ! 

Tested under: 
OS - Windows 10 
ESP-IDF: v5.1.1

1) 
Please, delete "/" because Python could not recognize the PATH 
image

2) "Browse" does not see .bin files 

image


3) Once project is flashed - there is only "Cancel" button is active.  This can be confusing for user. We should change this one. Maybe rename "Cancel" button add "Finish" button. 
image

4) Too much text there. I am not sure if it is needed. everything is intuitive here. 
image

**Comment by AndriiFilippov** [comment]
@sigmaaa hi ! 

Tested: 
OS - Windows 10 / Linux Ubuntu / MacOS 


LGTM 👍

**Review by AndriiFilippov — APPROVED** [review]

**Pull request merged**